### ISSUE TITLE
Possibly unneeded promise being returned from `initializers/setup-session-restoration`

### ISSUE THREAD

**Issue by shankarsridhar** [simplabs/ember-simple-auth#1750]
At least from `ember-simple-auth@1.8.0` and above, the [session.restore](https://github.com/simplabs/ember-simple-auth/blob/1.8.2/addon/initializers/setup-session-restoration.js#L16) is being called unconditionally, regardless of session being authenticated or not. And this is causing `beforeModel()` in my application to wait for promise to be resolved, even though, I don't do any async stuff in any of my application's `beforeModel` hooks. 

So as a result, this is causing unwanted loading template to be displayed, every time my route is refreshed.

Changing to the code from:

```
return session.restore().then(
  () => originalBeforeModel.apply(this, arguments),
  () => originalBeforeModel.apply(this, arguments)
);
```

to 

```
if (!this.session.isAuthenticated) {
  return session
    .restore()
    .then(
      () => originalBeforeModel.apply(this, arguments),
      () => originalBeforeModel.apply(this, arguments)
     );
} else {
  originalBeforeModel.apply(this, arguments);
}
```
fixes the issue for me.

In short, I return the promise from the beforeModel hook, only when the session is not authenticated, 
Is this the right way to fix this problem ? Happy to send a PR, if its correct

**Comment by marcoow** [comment]
`this.session.isAuthenticated` will always be `false` before `restore` has been called on the session. The `restore` method only reconstructs the session state based on the contents of the session store. It is only async because you **could** be using an async session store or the authenticator did anything async in its own `restore` method. If these 2 things are not the case for your app, the promise will resolve immediately actually.
